### DISCUSSION TITLE
How do I create/process 16bit png images properly?

### DISCUSSION THREAD

**Discussion by ghost** [ImageMagick/ImageMagick#6102] [Help] [answered]
### ImageMagick version

7.0.8-Q16-HDRI

### Operating system, version and so on

Windows 10

### Description

Hello. I rarely see a need to work in 16bit but I would likely benefit from it in this instance because I am doing a lot of image processing in IM involving gradients and blurs with PNG images. At the very least I would like to experiment to see if there is any noticeable difference.

I have read the documentation but Im not clear about how to tell IM to create/process 16bit images. Do I use -depth 16 or define png:format=png16? Also I noticed the png64: image path prefix but not sure if this is the same and sufficient on its own?

So for example, using a simple gradient as follows how would I make sure IM creates a 16bit gradient?

magick -size 30000x30000 gradient: linear_gradient.png

I appreciate any help with this.

**Comment by fmw42** [comment, marked as answer]
If you are using the default IM 7 install at Q16, then your PNG images created from scratch would be 16-bits automatically.  What do you get from 

magick -version

if it says Q16 (HDRI), then you have a 16-bit install.

**Reply by ghost** [reply]
Yes, it reads Q16 HDRI. I did a test with creating a gradient and opened the image in Photoshop and indeed its 16bit. I thought there was more to this. Thanks for pointing this out.

What exactly does PNG64: accomplish, for example:

magick input.png PNG64:output.png

**Comment by fmw42** [comment]
It forces output PNG to be 4 channels (RGBA) each at 16-bits so a total of 48-bits

**Reply by ghost** [reply]
Got it. Thank you.
